### PR TITLE
Add fsm comment

### PR DESCRIPTION
**Pull request by shes50103**
- Add comment on each state of FSM in lexer.go
- Rename initialize into initial

### CODE REVIEW

**Comment by shes50103** [comment]
I only add three lines comment to explain the purpose of states in FSM
Do I need to write more detail to  explain all the process in FSM?
Like how to enter each state and the effect after enter each state.

File: lexer/lexer.go [hunk 1]
```diff
@@ -21,10 +21,15 @@ func New(input string) *Lexer {
 	l.readChar()
 	l.FSM = fsm.NewFSM(
 		"initial",
+		/*
+			Initial state is default state
+			Nosymbol state support we identify tok ':' is symbol or json
```
Comment (st0012) [inline lexer/lexer.go]: I think this sentence could be more expressive if we change it to:
`"Nosymbol state helps us identify tok ':' is for symbol or hash value"`
(`{ key:value}` this is not a json, it's a hash)

File: lexer/lexer.go [hunk 2]
```diff
@@ -21,10 +21,15 @@ func New(input string) *Lexer {
...
+			Method state support we identify tok 'class' is kind of keyword or Ident
```
Comment (st0012) [inline lexer/lexer.go]: This should be changed to
`Method state helps us identify 'class' literal is a keyword or an identifier`

**Comment by st0012** [comment]
@shes50103 You don't need to explain whole process, but you can provide related document links in the comment.

**Comment by shes50103** [comment]
ok! I'll add the link of GitHub of FSM in comment

**Comment by st0012** [comment]
@shes50103 Thanks for adding comments

**Pull request merged**